### PR TITLE
[SPARK-590] Fixed hdfs-kerberos terasort spark test

### PR DESCRIPTION
**Pull request by susanxhuynh**
* Updated docker image version from `soak` to `latest`
* Updated hdfs service to foldered path
* Made the executors run as `root` for centos
* Added missing slash to "app name"
* Updated `hdfsclient` image version

### CODE REVIEW

**Comment by susanxhuynh** [comment]
cc @ArtRand A bunch of config adjustments to get the hdfs-kerberos spark job running.

File: tests/jobs/json/hdfs-kerberos-delete-terasort-files.json [hunk 1]
```diff
@@ -1,16 +1,12 @@
 {
   "id": "hdfs-kerberos-delete-terasort-files",
   "description": "Job that deletes Terasort files from Kerberized HDFS",
-  "labels": {},
   "run": {
     "cpus": 0.1,
     "mem": 512,
     "disk": 0,
-    "cmd": "/bin/bash && cd $MESOS_SANDBOX && python3 dcos-cli-auth.py --username=$DCOS_UID --password=$DCOS_PASSWORD && dcos task exec hdfsclient kinit -k -t hdfs.keytab hdfs/name-0-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL && dcos task exec hdfsclient bin/hdfs dfs -rm -r -f /terasort_in /terasort_out",
+    "cmd": "/bin/bash && cd $MESOS_SANDBOX && python3 dcos-cli-auth.py --username=$DCOS_UID --password=$DCOS_PASSWORD && dcos task exec hdfsclient kinit -k -t hdfs.keytab hdfs/name-0-node.data-serviceshdfs-kerberos.autoip.dcos.thisdcos.directory@LOCAL && dcos task exec hdfsclient bin/hdfs dfs -rm -r -f /terasort_in /terasort_out",
```
Comment (ArtRand) [inline tests/jobs/json/hdfs-kerberos-delete-terasort-files.json]: is there a way to template this? seems kinda fragile.

Comment (susanxhuynh) [reply tests/jobs/json/hdfs-kerberos-delete-terasort-files.json]: That's a good suggestion. I'll template it.

**Comment by susanxhuynh** [comment]
@ArtRand Templated the principal in the job config.

**Comment by ArtRand** [comment]
LGTM, I should probably parameterize the spark application name and package name for kerberos-Kafka right? We probably can't assume that there will always be a service named `/spark`?

**Comment by susanxhuynh** [comment]
That's a good idea. (parameterize spark app name and package name)

**Pull request merged**